types/client.d.ts

```diff
@@ -1,97 +1,218 @@
-import { FC } from "react"
+import * as React from "react"
 import { IncomingMessage } from "http"
-import { WithAdditionalParams } from "./_utils"
 import { Session } from "."
-import { AppProvider, DefaultProviders, Providers } from "./providers"
+import { BuiltInProviders, ProviderType } from "./providers"
 
-interface ContextProviderProps {
-  session: WithAdditionalParams<Session> | null | undefined
-  options?: SetOptionsParams
+export interface CtxOrReq {
+  req?: IncomingMessage
+  ctx?: { req: IncomingMessage }
 }
 
-interface SetOptionsParams {
-  baseUrl?: string
-  basePath?: string
-  clientMaxAge?: number
-  keepAlive?: number
+/***************
+ * Session types
+ **************/
+
+export type GetSessionOptions = CtxOrReq & {
+  event?: "storage" | "timer" | "hidden" | string
+  triggerEvent?: boolean
+}
+
+/**
+ * React Hook that gives you access
+ * to the logged in user's session data.
+ *
+ * [Documentation](https://next-auth.js.org/getting-started/client#usesession)
```

Comment:
Agreed! I checked, and there is nothing called `@docs` in JSDoc, so I rather used markdown, as it looks better :nail_care:. I went ahead, and added JSDoc comments to ALL option properties in `NextAuthOptions`, with additional links to documentation and other resources, example code snippets, and warnings where appropriate.

Hopefully, this will increase the productivity of people a lot, by not requiring them to even leave their IDEs for basic documentation.

Reply:
![image](https://user-images.githubusercontent.com/18369201/115232621-36c2fc00-a117-11eb-918d-545ac9a48479.png)
![image](https://user-images.githubusercontent.com/18369201/115232641-3e82a080-a117-11eb-9361-1090e8a0d91a.png)